lib/irb/ruby-lex.rb

```diff
@@ -406,10 +426,20 @@ def process_indent_level(tokens, lines, line_index, is_newline)
     prev_open_token = prev_opens.last
     next_open_token = next_opens.last
 
+    # Calculates base indent for pasted code on the line where prev_open_token is located
+    # irb(main):001:1*   if a # base_indent is 2, indent calculated from tokens is 0
+    # irb(main):002:1*         if b # base_indent is 6, indent calculated from tokens is 2
+    # irb(main):003:0>           c # base_indent is 6, indent calculated from tokens is 4
+    if prev_open_token
+      base_indent = [0, indent_difference(lines, line_results, prev_open_token.pos[0] - 1)].max
```

Comment:
Yes. It could be negative.

```ruby
if 1 # diff = 0
  if 2 # diff = 0
  end
end
```

```ruby
    if 1 # diff = 4
      if 2 # diff = 4
      end
    end
```

```ruby
    if 1 # diff = 4
    if 2 # diff = 2
    end
    end
```

```ruby
if 1 # diff = 0
if 2 # diff = -2 ← negative
end
end
```